src/harness/sas_objects.py

```diff
@@ -0,0 +1,297 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import security_testcase
+from util import winnforum_testcase
+import json
+import os
+import sas
+import urllib
+
+class Grant:
+
+    '''
+	A Grant object has the following methods
+	Initialize
+   	   Parameters: grant ID, grant request
+	Methods
+	Retrieve internal information
+	    grant ID,
+	    grant request
+	Retrieve operating params
+	Construct heartbeat request
+	Update “was authorized in last heartbeat? state
+
+    '''
+
+    def __init__(self,grantId,grantRequest):
+        self.GrantId = grantId
+        self.GrantRequest = grantRequest
+        self.IsGrantAuthorisedInlastHeartbeat = False
+
+    def GetGrantId(self):
+        return self.GrantId
+
+    def GetGrantRequest(self):
+        return self.GrantRequest
+
+    def GetRequestOperationParam(self):
+        return self.GrantRequest['operationParam']
+
+    def ConstructHeartBeatRequest(self):
+
+        hearbeatRequest =  {
+        "cbsdId": self.GrantRequest['cbsdId'],
+        'grantId': self.GrantId,
+        'operationState': "GRANTED"
+        }
+        return hearbeatRequest
+
+    def UpdateGrantAuthorisedInlastHeartbeat(self,IsAuthorized):
+        IsGrantAuthorisedInlastHeartbeat = IsAuthorized
```

Comment:
This doesn't actually set the member variable.

Reply:
used the self.grant_authorized_in_last_heartbeat instead of grant_authorized_in_last_heartbeat